src/main/java/com/easypost/easyvcr/interactionconverters/HttpUrlConnectionInteractionConverter.java

```diff
@@ -0,0 +1,117 @@
+            String uriString = connection.getURL().toString();
+            Map<String, List<String>> headers = connection.getRequestProperties();
+            String method = connection.getRequestMethod();
+
+            // apply censors
+            uriString = censors.applyQueryParametersCensors(uriString);
+            headers = censors.applyHeadersCensors(headers);
+
+            // create the request
+            Request request = new Request();
+            request.setMethod(method);
+            request.setUri(new URI(uriString));
+            request.setHeaders(headers);
+            // TODO: Capture and censor body
+            // request.setBody(body);
+
+            return request;
+        } catch (URISyntaxException ignored) {
+        }
+        // FIXME: make it so this won't blow up on error
+        return null;
+    }
+
+    /**
+     * Convert a HttpURLConnection response to a ResponseAndTime object.
+     * @param connection The HttpURLConnection response.
+     * @param censors The censors to apply to the response.
+     * @return The ResponseAndTime object.
+     */
+    public ResponseAndTime createRecordedResponse(HttpURLConnection connection, Censors censors) {
+        try {
+            // quickly time how long it takes to get the initial response
+            Instant start = Instant.now();
+            int responseCode = connection.getResponseCode();
+            Instant end = Instant.now();
+            long milliseconds = Duration.between(start, end).toMillis();
+
+            // collect elements from the connection
+            String message = connection.getResponseMessage();
+            String uriString = connection.getURL().toString();
+            Map<String, List<String>> headers = connection.getHeaderFields();
+            String body = null;
+            String errors = null;
+            try {
+                body = readFromInputStream(connection.getInputStream());
+                errors = readFromInputStream(connection.getErrorStream());
+            } catch (NullPointerException ignored) {
+            }
+
+            // apply censors
+            uriString = censors.applyQueryParametersCensors(uriString);
+            headers = censors.applyHeadersCensors(headers);
+            // we don't censor the response body, only the request body
+
+            // create the response
+            Response response = new Response();
+            response.setStatus(new Status(responseCode, message));
+            response.setUri(new URI(uriString));
+            response.setHeaders(headers);
+            if (body != null) {
+                response.setBody(body);
+            }
+            if (errors != null) {
+                response.setErrors(errors);
+            }
+
+            return new ResponseAndTime(response, milliseconds);
+        } catch (URISyntaxException | IOException ignored) {
+        }
+        // FIXME: make it so this won't blow up on error
```

Comment:
Ultimately parses the same way ([an IDE finds it the same way it finds a TODO](https://docs.microsoft.com/en-us/visualstudio/mac/task-comments?view=vsmac-2019)), but okay.

Reply:
It's dangerous to assume the the magic features of any one IDE is how everyone will access or use something. EasyPost is 99% terminal based editors internally (our team is the exception) so although it may work in yours, it doesn't work in others. Certainly if I go on a `TODO fix spree`, I'll miss anything that's titled `FIXME` because I'm only searching for TODO. We also use `TODO` almost exclusively internally for that reason.